.github/workflows/ci.yml

```diff
@@ -21,54 +21,28 @@ env:
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
 jobs:
+  setup: 
+    runs-on: ubuntu-22.04
+    outputs:
+      matrix: ${{ steps.read-json.outputs.matrix }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Read JSON
+        id: read-json
```

Comment:
The name is a little counter-intuitive to me since this job is writing JSON to the output